src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -494,7 +498,7 @@ void SolverInterfaceImpl::finalize()
   Event                    e("finalize"); // no precice::syncMode here as MPI is already finalized at destruction of this event
   utils::ScopedEventPrefix sep("finalize/");
 
-  if (_state == State::Initialized) {
```

Comment:
This has to be both `State::Initialized` and `State::InitializedData`.
We establish communication channels in `initialize()`

Reply:
Locally I added some more checks at the beginning of `finalize()`:
```
void SolverInterfaceImpl::finalize()
{
  PRECICE_TRACE();
  PRECICE_CHECK(_state != State::Finalized, "finalize() may only be called once.")
  PRECICE_CHECK(_state != State::Initialized, "initializeData() has to be called before finalize().");
  PRECICE_CHECK(_state == State::InitializedData, "initializeData() has to be called before advance().")
  ...
```
This asserts that `initialize()` and `initializeData()` have been called before `finalize()`. This is an additional restriction that I would like to introduce here, because I don't really see a use-case for calling `finalize()` without `initialize()`. Does anybody know why we currently support this?

Reply:
Fixed it in [4ff2cbe](https://github.com/precice/precice/pull/1343/commits/4ff2cbe7d7e0ba3b89009ae32274a66295b58bcb).